tiledb/py_arrow_io_impl.h

```diff
@@ -778,8 +777,21 @@ void ArrowExporter::export_(const std::string &name, ArrowArray *array,
       // take the size of the entire buffer and divide by the size of each
       // element
       elem_num = bufferinfo.data_num / bufferinfo.tdbtype.cell_val_num;
-    } else {
-      elem_num = bufferinfo.data_num;
+    } else if (arrow_fmt.fmt_ == "tdD") {
+      // for Arrow date32 we only need the first 4 bytes of each 8-byte
+      // TILEDB_DATETIME_DAY element which we keep by in-place left shifting
```

Comment:
Isn't such an approach going to bite us if we ever run on a non-little-endian machine and the ordering is different?   Or am I misreading this?